src/activity_item_handling.cpp

```diff
@@ -1992,12 +1991,12 @@ static void fetch_activity( player &p, const tripoint &src_loc,
                     }
                     it.set_var( "activity_var", p.name );
                     p.i_add( it );
-                    picked_up = it.tname();
                     items_there.erase( item_iter );
                     // If we didn't pick up a whole stack, put the remainder back where it came from.
                     if( leftovers.charges > 0 ) {
                         g->m.add_item_or_charges( src_loc, leftovers );
                     }
+                    std::string picked_up = it.tname();
                     if( p.is_npc() && !picked_up.empty() ) {
```

Comment:
Under what circumsntance would `item::tname` return an empty string? It seems this check will never be false.